src/execute.ts

```diff
@@ -292,9 +296,14 @@ export default function executeFirestoreNodes({
       nodeParentSnap,
     })
     if (operationType === 'query') {
-      resolver(ref).then(handleValue, onError)
+      if (ref.type === 'document') {
+        getDoc(ref).then(handleValue)
+      } else {
+        getDocs(ref).then(handleValue)
+      }
     } else {
-      const unlisten = onSnapshot(ref, handleValue, onError)
+      const unlisten =
+        ref.type === 'document' ? onSnapshot(ref, handleValue, onError) : onSnapshot(ref, handleValue, onError)
```

Comment:
What's the difference between the two conditions here?

Reply:
`onSnapshot` is actually overloaded based on the type of the parameters, so the `ref.type` check helps typescript know which `onSnapshot` is being used - does that make sense?